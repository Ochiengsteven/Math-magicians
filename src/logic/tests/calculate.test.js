import calculate from '../calculate';
// eslint-disable-next-line no-unused-vars
import operate from '../operate';

describe('calculate function', () => {
  it('resets state on AC button press', () => {
    const initialState = {
      total: '123',
      next: '456',
      operation: '+',
    };
    const newState = calculate(initialState, 'AC');

    expect(newState).toMatchSnapshot();
  });

  it('calculates the result on = button press', () => {
    const initialState = {
      total: '10',
      next: '5',
      operation: '+',
    };
    const newState = calculate(initialState, '=');

    expect(newState).toMatchSnapshot();
  });

  it('handles +/- button press', () => {
    const initialState = {
      total: '15',
      next: '5',
      operation: null,
    };

    const operation = '+/-';
    const newState = calculate(initialState, operation);

    expect(newState).toMatchSnapshot();
  });
});
