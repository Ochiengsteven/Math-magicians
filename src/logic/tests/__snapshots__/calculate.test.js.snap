// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate function calculates the result on = button press 1`] = `
Object {
  "next": null,
  "operation": "=",
  "total": "15",
}
`;

exports[`calculate function handles +/- button press 1`] = `
Object {
  "next": "-5",
  "operation": null,
  "total": "15",
}
`;

exports[`calculate function resets state on AC button press 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": null,
}
`;
